app/scripts/controllers/createPersistentVolumeClaim.js

```diff
@@ -55,29 +55,34 @@ angular.module('openshiftConsole')
           }
         };
 
-      function generatePersistentVolumeClaim() {
-        var pvc = {
-          kind: "PersistentVolumeClaim",
-          apiVersion: "v1",
-          metadata: {
-            name: $scope.claim.name,
-            labels: {}
-          },
-          spec: {
-            resources: {
-              requests:{}
-
+         function generatePersistentVolumeClaim() {
+          var pvc = {
+            kind: "PersistentVolumeClaim",
+            apiVersion: "v1",
+            metadata: {
+              name: $scope.claim.name,
+              labels: {}
+            },
+            spec: {
+              resources: {
+                requests:{}
+              },
+              selector:{
+                matchLabels:{}
+              }
             }
-          }
-        };
-
-        pvc.spec.accessModes = [$scope.claim.accessModes || "ReadWriteOnce"] ;
-        var unit =  $scope.claim.unit || "Mi";
-        pvc.spec.resources.requests.storage = $scope.claim.amount + unit;
-
-        return pvc;
-      }
+          };
 
+          pvc.spec.accessModes = [$scope.claim.accessModes || "ReadWriteOnce"] ;
+          var unit =  $scope.claim.unit || "Mi";
+          pvc.spec.resources.requests.storage = $scope.claim.amount + unit;
+          _.each($scope.claim.selectedLabels, function(label) {
+            if (label.name !== "" && label.value !== "") {
```

Comment:
Right! That should be all that is needed to cleanup the list, there is also `.mapEntries()` to convert from list to object, I believe this will do all of that:
```javascript
pvc.spec.selector.matchLabels = keyValueEditorUtils.mapEntries( keyValueEditorUtils.compactEntries($scope.claim.selectedLabels) );
```